src/net.cpp

```diff
@@ -1222,6 +1222,13 @@ void CConnman::DisconnectNodes() {
     }
 }
 
+void CConnman::InactivityChecks() {
+    LOCK(cs_vNodes);
+    for (CNode* pnode : vNodes) {
+        InactivityCheck(pnode);
```

Comment:
Commit "Move InactivityChecks logic to private method."

Just noting that with this change `InactivityCheck` is now called with `cs_vNodes` locked.

Reply:
The reasoning here is that the contents of InactivityCheck are virtually instant.

If this causes any issues in the future it can be trivially switched back to using reference counting.